location/src/commonTest/kotlin/location/LocationStateTest.kt

```diff
@@ -0,0 +1,574 @@
+            }
+            test {
+                assertTrue(it is LocationState.Enabled)
+                assertEquals(location1, it.location)
+            }
+        }
+        permissionManager.hasStoppedMonitoring.await()
+        locationManager.stopMonitoringPermissionsCompleted.await()
+        locationManager.stopMonitoringLocationCompleted.await()
+        locationManager.stopMonitoringLocationEnabledCompleted.await()
+
+        permissionManager.reset()
+        locationManager.reset()
+        permissionManager.currentState = PermissionState.Denied.Locked(permissionManager)
+        FlowTest(locationStateRepo.flowable).testWithFlow {
+            test {
+                assertTrue(it is LocationState.Disabled.NotPermitted)
+                assertEquals(Location.UnknownLocation.WithLastLocation(location1, Location.UnknownLocation.Reason.PERMISSION_DENIED), it.location)
+            }
+        }
+
+        permissionManager.hasStoppedMonitoring.await()
+        locationManager.stopMonitoringPermissionsCompleted.await()
+    }
+
+    @Test
+    fun testResumeFlowGPSDisabled() = runBlocking {
+        setupLocationState(Permission.Location(background = false, precise = false), autoRequestPermission = false, autoEnableLocations = false)
+        permissionManager.currentState = PermissionState.Allowed(permissionManager)
+        locationManager.locationEnabled = true
+        FlowTest(locationStateRepo.flowable).testWithFlow {
+            test {
+                assertTrue(it is LocationState.Enabled)
+                assertEquals(Location.UnknownLocation.WithoutLastLocation(Location.UnknownLocation.Reason.NOT_CLEAR), it.location)
+            }
+            action {
+                locationManager.handleLocationChanged(location1)
+            }
+            test {
+                assertTrue(it is LocationState.Enabled)
+                assertEquals(location1, it.location)
+            }
+        }
+        permissionManager.hasStoppedMonitoring.await()
+        locationManager.stopMonitoringPermissionsCompleted.await()
+        locationManager.stopMonitoringLocationCompleted.await()
+        locationManager.stopMonitoringLocationEnabledCompleted.await()
+        permissionManager.reset()
+        locationManager.reset()
+        locationManager.locationEnabled = false
+        FlowTest(locationStateRepo.flowable).testWithFlow {
+            test {
+                assertTrue(it is LocationState.Disabled.NoGPS)
+                assertEquals(Location.UnknownLocation.WithLastLocation(location1, Location.UnknownLocation.Reason.NO_GPS), it.location)
+            }
+        }
+        permissionManager.hasStoppedMonitoring.await()
+        locationManager.stopMonitoringPermissionsCompleted.await()
+        locationManager.stopMonitoringLocationEnabledCompleted.await()
+        // This test seems to not clean itself properly causing issues with other tests. No idea why so some delay
+        delay(500)
```

Comment:
I have no idea why, but this test does not clean properly, causing other tests to fail
Ive added a delay to solve it for now but obviously something is wrong, so if someone sees the reason please let me know @thoutbeckers @dasdranagon @avdyushin